stock_storage_type/models/stock_location.py

```diff
@@ -0,0 +1,652 @@
+        recursive=True,
+    )
+    # TODO: Maybe renaming these fields as there are already such fields
+    # in core but without domains. Something like 'pending_in_move_ids'
+    in_move_ids = fields.One2many(
+        "stock.move",
+        "location_dest_id",
+        domain=[
+            ("state", "in", ("waiting", "confirmed", "partially_available", "assigned"))
+        ],
+        help="technical field: the pending incoming stock.moves in the location",
+    )
+
+    in_move_line_ids = fields.One2many(
+        "stock.move.line",
+        "location_dest_id",
+        domain=[
+            ("state", "in", ("waiting", "confirmed", "partially_available", "assigned"))
+        ],
+        help="technical field: the pending incoming "
+        "stock.move.lines in the location",
+    )
+    out_move_line_ids = fields.One2many(
+        "stock.move.line",
+        "location_id",
+        domain=[
+            ("state", "in", ("waiting", "confirmed", "partially_available", "assigned"))
+        ],
+        help="technical field: the pending outgoing "
+        "stock.move.lines in the location",
+    )
+    location_will_contain_lot_ids = fields.Many2many(
+        "stock.lot",
+        store=True,
+        compute="_compute_location_will_contain_lot_ids",
+        help="technical field: list of stock.lots in "
+        "the location, either now or in pending operations",
+    )
+    location_will_contain_product_ids = fields.Many2many(
+        "product.product",
+        store=True,
+        compute="_compute_location_will_contain_product_ids",
+        help="technical field: list of products in "
+        "the location, either now or in pending operations",
+    )
+
+    leaf_location_ids = fields.Many2many(
+        "stock.location",
+        recursive=True,
+        compute="_compute_leaf_location_ids",
+        help="technical field: all the leaves locations",
+    )
+    leaf_child_location_ids = fields.Many2many(
+        "stock.location",
+        recursive=True,
+        compute="_compute_leaf_location_ids",
+        help="technical field: all the leaves sub-locations",
+    )
+    max_height = fields.Float(
+        related="computed_storage_category_id.max_height",
+        store=True,
+        recursive=True,
+    )
+    max_height_in_m = fields.Float(
+        related="computed_storage_category_id.max_height_in_m",
+        store=True,
+        recursive=True,
+    )
+
+    do_not_mix_products = fields.Boolean(
+        compute="_compute_do_not_mix_products", store=True, recursive=True
+    )
+    do_not_mix_lots = fields.Boolean(
+        compute="_compute_do_not_mix_lots", store=True, recursive=True
+    )
+    only_empty = fields.Boolean(
+        compute="_compute_only_empty", store=True, recursive=True
+    )
+
+    @api.depends(
+        "usage",
+        "computed_storage_category_id.do_not_mix_lots",
+        "computed_storage_category_id.capacity_ids.do_not_mix_lots",
+    )
+    def _compute_do_not_mix_lots(self):
```

Comment:
@jbaudoux So, the last thing is to move this to storage category.

As there is a manual field there too. Do you see an approach to allow setting a value on the category itself apart the capacities (colliding field names) ?

Reply:
@jbaudoux